ibis/expr/operations/core.py

```diff
@@ -17,8 +16,23 @@ def distinct_roots(*expressions):
     return list(toolz.unique(roots))
 
 
+def _compare_items(a, b):
+    if hasattr(a, "equals"):
```

Comment:
Just try the call here, it's cheaper than checking if the attribute exists, and in the case it doesn't the cost is the same because `hasattr` uses EAFP as well.